src/Router/Artsy2.tsx

```diff
@@ -0,0 +1,48 @@
+import { ContextProps as _ContextProps } from "Components/Artsy"
+import React, { SFC } from "react"
+import { getUser } from "Utils/getUser"
+import { createEnvironment } from "../Relay/createEnvironment"
+
+// FIXME: Move from Artsy to Artsy2
+export interface ContextProps extends _ContextProps {
+  // FIXME: Type properly
+  mediator?: any
+  system?: any
+}
+
+const Context = React.createContext<ContextProps>({})
+
+/**
+ * Creates a new Context.Provider with a user and Relay environment, or defaults
+ * if not passed in as props.
+ */
+export const ContextProvider: SFC<ContextProps> = ({ children, ...props }) => {
+  const currentUser = getUser(props.currentUser)
+  const relayEnvironment =
+    props.relayEnvironment || createEnvironment({ user: currentUser })
+
+  const providerValues = {
+    ...props,
+    currentUser,
+    relayEnvironment,
+  }
+
+  return <Context.Provider value={providerValues}>{children}</Context.Provider>
+}
+
+export const ContextConsumer = Context.Consumer
```

Comment:
Is this missing typings for the callsite of the consumer or do they come in on line 13?

Reply:
This is passed through fr line 13: 

<img width="403" alt="screen shot 2018-08-23 at 12 56 40 pm" src="https://user-images.githubusercontent.com/236943/44548813-03b0dc00-a6d4-11e8-8941-77ab471f3380.png">